Update of the abbreviations

federicapicogna opened the pull request.

federicapicogna committed:
Update abbreviations

koenderks reviewed (changes requested):
1. Didn't we settle on negative predictive rate parity? Why the change **back** to negative predictive value? I think you might not have reset your development branch after merging your previous pull request.
2. What happens when you use `metric = "eo"` in `model_fairness()`. It should at least give a warning message or something that it is not possible.
3. Can you make sure that the unit tests work again.
4. Change "dp" back to "pp" since the proportional parity is equivalent to the disparate impact. "dp" is demographic parity and is calculated as (TP + FP)

federicapicogna committed:
Update

Predicted changed into predictive,
confusion matrix inverted,
disparate impact is pp and not dp,
missing = error for eo

federicapicogna committed:
matrix going back

federicapicogna committed:
Predictive instead of predicted

koenderks commented:
Unit tests still fail:

```
Failure ('test-fairness-selection.R:25:3'): Validation of fairness selection ──
outcome$measure not equal to "dp".
1/1 mismatches
x[1]: "eo"
y[1]: "dp"
```

koenderks reviewed (changes requested):
Check the unit tests

federicapicogna committed:
Test update

koenderks commented:
I think this still needs the error handing for metric="eo", right?

federicapicogna committed:
Legend update

federicapicogna committed:
update legend

federicapicogna committed:
Update

federicapicogna committed:
Update

koenderks commented:
This seems like a lot of empty space, can you put the legend closer to the diagram? Also, revert the changes in the `stanmodels.R` file and don't forget to implement error handling for `metric = "eo"` in `model_fairness()`

![image](https://github.com/user-attachments/assets/636eda00-783a-4675-8cbf-d2502c7fc729)

federicapicogna commented:
The strange thing about stanmodels is that when I run the lint test, it tells me that I have to change it this way or the test will fail. As for the rest, I fixed the legend and I'm now working on "eo". 
<img width="706" alt="Screenshot 2025-02-06 at 10 13 52" src="https://github.com/user-attachments/assets/7403eb16-ef03-49d6-b6fa-d494cd593f05" />

federicapicogna committed:
Legend position

koenderks commented:
Just implement a simple error message for when metric is "eo", we can handle the rest in JASP. Then undo all changes in stanmodels.R

koenderks reviewed (changes requested):
See comment above

federicapicogna committed:
Error message for eo

koenderks commented on R/model_fairness.R:
```diff
@@ -238,181 +241,185 @@ model_fairness <- function(data,
-  if (metric != "dp") {
+  if (metric == "eo") {
+    stop("Error: 'Equalized Odds' (EO) requires simultaneous evaluation of 'True Positive Rate' (TPR) and 'False Positive Rate' (FPR). You must first select tprp, then fprp, and ensure both fairness measures conclude no discrimination for both privileged and unprivileged groups before concluding no discrimination according to EO.")
+  } else {
```
Remove the `else` here, you quit the analysis at this point anyway if metric = "eo"

koenderks reviewed (changes requested):
There are lots of changes in the `model_fairness.R` file, but you only need to add an error message. Please make sure that the changes here are (or show up in the pull request as) minimal.

federicapicogna committed:
Removing else

federicapicogna committed:
Stan
